boards/xtensa/intel_adsp_cavs25/doc/index.rst

```diff
@@ -0,0 +1,409 @@
+
+By default neither the "chronos" user nor root accounts have
+passwords, so unless you want to type a ssh key in by hand, you
+probably want to set a password for the first login (before you run
+ssh-copy-id, of course):
+
+.. code-block:: console
+
+    cros# passwd
+
+Now ssh into the chromebook and add your key to
+``.ssh/authorized_keys`` as you do for any Linux system.
+
+Install Crouton
+***************
+
+The Zephyr integration tools require a proper Linux environment and
+won't run on ChromeOS's minimal distro.  So we need to install a Linux
+personality.  **DO NOT** bother installing the "Linux Development
+Environment" (Crostini) from the ChromeOS Developer settings.  This
+personality runs inside a VM, where our tools need access to the real
+kernel running on the real hardware.  Instead install Crouton
+(https://github.com/dnschneid/crouton), which is a community
+chroot-based personality that preserves access to the real hardware
+sysfs and /dev filesystem.  These instructions install the "cli-extra"
+package list, there are X11-enabled ones available too if you prefer
+to work on the device screen directly.  See the project page, etc...
+
+At a root shell, grab the installer and run it (note: /usr/local is
+the only writable filesystem without noexec, you must place the binary
+there for it to run!):
+
+.. code-block:: console
+
+    cros# mkdir -p /usr/local/bin
+    cros# curl -L https://github.com/dnschneid/crouton/raw/master/installer/crouton \
+                  > /usr/local/bin/crouton
+    cros# chmod 755 /usr/local/bin/crouton
+    cros# crouton -r focal -t cli-extra
+
+Start the Crouton chroot environment:
+
+.. code-block:: console
+
+    cros# startcli
+
+Now you are typing commands into the Ubuntu environment.  Enable
+inbound ssh on Crouton, but on a port other than 22 (which is used for
+the native ChromeOS ssh server).  I'm using 222 here (which is easy to
+remember, and not a registered port in /etc/services):
+
+.. code-block:: console
+
+    crouton# apt install iptables openssh-server
+    crouton# echo "Port 222" >> /etc/ssh/sshd_config
+    crouton# mkdir /run/sshd
+    crouton# iptables -I INPUT -p tcp --dport 222 -j ACCEPT
+    crouton# /usr/sbin/sshd
+
+(*As above: note that we have introduced a hostname of "crouton" to
+refer to the separate Linux personality.*)
+
+NOTE: the mkdir, iptables and sshd commands need to be run every time
+the chroot is restarted.  You can put them in /etc/rc.local for
+convenience.  Crouton doesn't run systemd (because it can't -- it
+doesn't own the system!) so Ubuntu services like openssh-server don't
+know how to start themselves.
+
+Building and Installing a Custom Kernel
+***************************************
+
+On your build host, grab a copy of the ChromeOS kernel tree.  The
+shipping device is using a 5.4 kernel, but the 5.10 tree works for me
+and seems to have been backporting upstream drivers such that its main
```

Comment:
You could add a remark to explain why this is needed. I guess the only mandatory req is the hugetbl support.

Reply:
That's right.  And in fact we don't strictly need hugetlbfs.  Transparent huge pages would work too (and are supported by default in chromeos kernels), I was just a little confused about the rules for when they're provided vs. when you get 4k pages as a fallback, and wasn't completely sure how to detect them from the /proc pagemap file.  The filesystem has simpler semantics.

And even then, we don't actually NEED the 2M page.  With only 4k pages, you can set up a DMA stream with 1M of buffer, which is enough to store (hopefully) any firmware we want to push.  But setting up scatter gather DMA from userspace seemed too difficult vs. just using a known-contiguous region. :)

Reply:
But it's true: without the hugetlbfs need, Zephyr itself probably would be fine with the factory 5.4 kernel.  I guess I'm assuming that all these devices are going to want to be running current SOF kernel drivers at some point, since at the end of the day that's the only app we truly need to deploy.